toolkit/tools/internal/ccachemanager/ccachemanager.go

```diff
@@ -0,0 +1,552 @@
+				groupSize = len(group.PackageNames)
+				break
+			}
+		}
+		if groupName != "" {
+			break
+		}
+	}
+
+	if groupName == "" {
+		logger.Log.Infof("  did not find ccache group for (%s) - assigning to group \"%s\".", basePackageName, CommonGroupName)
+		groupName = CommonGroupName
+		groupSize = 0
+	}
+
+	return groupName, groupSize
+}
+
+func (m *CCacheManager) findCCacheGroupSize(groupName string) (groupSize int) {
+
+	groupSize = 0
+
+	for _, group := range m.Configuration.Groups {
+		if groupName == group.Name {
+			groupSize = len(group.PackageNames)
+		}
+	}
+
+	return groupSize
+}
+
+func (m *CCacheManager) getPkgCCacheDir(pkgCCacheGroupName string, pkgArchitecture string) (string, error) {
+	if pkgArchitecture == "" {
+		return "", errors.New("CCache package pkgArchitecture cannot be empty.")
+	}
+	if pkgCCacheGroupName == "" {
+		return "", errors.New("CCache package group name cannot be empty.")
+	}
+	return m.RootCCacheDir + "/" + pkgArchitecture + "/" + pkgCCacheGroupName, nil
+}
+
+func (m *CCacheManager) DownloadPkgGroupCCache() (err error) {
+
+	logger.Log.Infof("** processing download of ccache artifacts...")
+
+	if m.PkgGroupName == CommonGroupName {
+		logger.Log.Infof("  %s group - skipping download...", CommonGroupName)
+		return nil
+	}
+
+	remoteStoreConfig := m.Configuration.RemoteStoreConfig
+	if !remoteStoreConfig.DownloadEnabled {
+		logger.Log.Infof("  downloading archived ccache artifacts is disabled. Skipping download...")
+		return nil
+	}
+
+	logger.Log.Infof("  downloading and expanding...")
+	err = ensureDirExists(m.DownloadsDir)
+	if err != nil {
+		logger.Log.Warnf("Cannot create ccache download folder. Error: %v", err)
+		return err
+	}
+
+	logger.Log.Infof("  creating container client...")
+	azureblobstorage, err := azureblobstoragepkg.Create(remoteStoreConfig.TenantId, remoteStoreConfig.UserName, remoteStoreConfig.Password, remoteStoreConfig.StorageAccount, azureblobstoragepkg.AnonymousAccess)
+	if err != nil {
+		logger.Log.Warnf("Unable to init azure blob storage client. Error: %v", err)
+		return err
+	}
+
+	if remoteStoreConfig.DownloadFolder == LatestTagMarker {
+
+		logger.Log.Infof("  ccache is configured to use the latest...")
+
+		// Download the tags file...
+		logger.Log.Infof("  downloading (%s) to (%s)...", m.PkgTagFile.RemoteSourcePath, m.PkgTagFile.LocalSourcePath)
+		err = azureblobstorage.Download(context.Background(), remoteStoreConfig.ContainerName, m.PkgTagFile.RemoteSourcePath, m.PkgTagFile.LocalSourcePath)
+		if err != nil {
+			logger.Log.Warnf("  unable to download ccache archive. Error: %v", err)
+			return err
+		}
+
+		// Read the text contents...
+		latestBuildTag, err := ioutil.ReadFile(m.PkgTagFile.LocalSourcePath)
+		if err != nil {
+			logger.Log.Warnf("Unable to read ccache tag file contents. Error: %v", err)
+			return err
+		}
+
+		// Adjust the download folder from 'latest' to the tag loaded from the file...
+		remoteStoreConfig.DownloadFolder = string(latestBuildTag) 
```

Comment:
This logic needs to be update. Updating the DownloadFolder has no effect on the download path at this point (it used to in a previous code iteration). Working on a fix now and will update the PR.